src/Functions/HasTokenImpl.h

```diff
@@ -43,34 +44,51 @@ struct HasTokenImpl
         const UInt8 * const end = haystack_data.data() + haystack_data.size();
         const UInt8 * pos = begin;
 
-        /// The current index in the array of strings.
-        size_t i = 0;
-
-        TokenSearcher searcher(pattern.data(), pattern.size(), end - pos);
-
-        /// We will search for the next occurrence in all rows at once.
-        while (pos < end && end != (pos = searcher.search(pos, end - pos)))
+        try
         {
-            /// Let's determine which index it refers to.
-            while (begin + haystack_offsets[i] <= pos)
+            /// Parameter `pattern` is supposed to be a literal of letters and/or numbers.
+            /// Otherwise, an exception from the constructor of `TokenSearcher` is thrown.
+            /// If no exception is thrown at that point, then no further error cases may occur.
+            TokenSearcher searcher(pattern.data(), pattern.size(), end - pos);
+            if (res_null)
+                std::ranges::fill(res_null->getData(), false);
+
+            /// The current index in the array of strings.
+            size_t i = 0;
+            /// We will search for the next occurrence in all rows at once.
+            while (pos < end && end != (pos = searcher.search(pos, end - pos)))
             {
-                res[i] = negate;
+                /// Let's determine which index it refers to.
+                while (begin + haystack_offsets[i] <= pos)
+                {
+                    res[i] = negate;
+                    ++i;
+                }
+
+                /// We check that the entry does not pass through the boundaries of strings.
+                if (pos + pattern.size() < begin + haystack_offsets[i])
+                    res[i] = !negate;
+                else
+                    res[i] = negate;
+
+                pos = begin + haystack_offsets[i];
                 ++i;
             }
 
-            /// We check that the entry does not pass through the boundaries of strings.
-            if (pos + pattern.size() < begin + haystack_offsets[i])
-                res[i] = !negate;
+            /// Tail, in which there can be no substring.
+            if (i < res.size())
+                memset(&res[i], negate, (res.size() - i) * sizeof(res[0]));
+        }
+        catch (...)
```

Comment:
@rschu1ze @kssenii 

I'm surprised by this code.
It is obviously incorrect, it is slow, and record errors in the system.errors table.

Reply:
@alexey-milovidov the range you highlighted is mostly unchanged. Or is it the try-catch as @rschu1ze is suggesting? If you could provide some way of reproducing potential issues with it I'm happy to contribute an improvement.

Reply:
Yes, `catch (...)` is the problem.